doc/parent_child_spec.mld

```diff
@@ -157,3 +161,35 @@ and we shall get
 ]}
 
 For more about [odoc] commands, simply invoke [odoc --help] in your shell.
+
+{1 Convention for installed packages}
+
+Locally, the build system can make arbitrary complex documentation page
+hierarchies.
+However, the generated HTML documentation is generally not installed as part of
+a package. Instead the documentation source code made of [.mld] pages is
+installed and might be used by a different driver.
+
+{2 Convention}
+
+In order for drivers to build consistent documentation for a package, the
+following convention should be followed.
+
+- [.mld] pages are installed in a package's [share] directory, under the
+  [odoc-pages] sub-directory.
+- A page is the parent of every installed pages. The driver can freely name this
+  page, for example it can be named after the package. In what follows, we
+  refer to this page as the [pkg] page.
+- If there is an installed [index.mld] file, the driver has to use it as
+  content for the [pkg] page.
+- If there is no installed [index.mld] page, the driver has to generate some
+  content for the [pkg] page.
+
+When the rendering of source code is enabled, the source tree will be named
```

Comment:
I don't think any driver that follows this spec renders source today. If we're just documenting how things work right now we should omit this section.

Reply:
`doc/driver.mld` does that. This document doesn't describe what an existing driver does, rather what can be done by a driver today.

Reply:
I think it's better if we document what driver should do, so that there are no useless discrepancies between them.

Reply:
I think this paragraph should be kept then.

Reply:
doc/driver.mld is a bad example for this - it follows literally none of the other conventions in the paragraph, as the intent back when it was first written was to show how odoc can be used more generally than just for making docs from installed opam packages.

We haven't quite figured out a convention for the sources, and it's not at all clear to me that the hierarchy will share the same namespace as the other package pages. As a concrete example, on `ocaml.org` cmdliner's pages currently appear under the hierarchy https://ocaml.org/p/cmdliner/latest/doc/ - e.g. tutorial.mld is https://ocaml.org/p/cmdliner/latest/doc/tutorial.html . I can easily imagine that there would be a desire for the source to appear under https://ocaml.org/p/cmdliner/latest/src/ in which case there's no namespace clash.

I'd quite like in this PR to just focus on what's actually implemented 'in the wild' and then we can have a separate discussion on these sorts of topics to bash out the wider question about what we should do to support the new features.

Reply:
I don't agree that what exists in the wild is in the scope of this PR but I removed the paragraph to not interferes with the future changes.